elements/lisk-chain/src/chain.ts

```diff
@@ -432,26 +412,27 @@ export class Chain {
 			removeFromTempTable: false,
 		},
 	): Promise<void> {
-		return this.storage.entities.Block.begin('saveBlock', async tx => {
```

Comment:
Could you describe how this is guaranteed with the new store?

Reply:
This will be solved by `batch` write.
when we want to atomically save the data,
```
const batch = this._db.batch();
batch.put('xx', xxx);
batch.del('xx', xxx);
await batch.write();
```
and here `put` and `del` for batch are sync